src/utils/slpParser.ts

```diff
@@ -1,108 +1,104 @@
-    }
-
-    public getFrames(): FramesType | null {
-        return this.playerFrames;
-    }
-
-    public getFollowerFrames(): FramesType | null {
-        return this.followerFrames;
-    }
-
-    public handleGameEnd(payload: GameEndType): void {
-        payload = payload as GameEndType;
-        this.gameEnd = payload;
-    }
-
-    public handleGameStart(payload: GameStartType): void {
-        if (!payload.stageId) {
-            return;
-        }
-        this.settings = payload;
-        const players = payload.players;
-        this.settings.players = players.filter(player => player.type !== 3);
-        this.opponentIndices = getSinglesOpponentIndicesFromSettings(this.settings);
-        this.statsComputer.setOpponentIndices(this.opponentIndices);
-    }
-
-    public handlePostFrameUpdate(payload: PostFrameUpdateType): void {
-        if (payload.frame === null) {
-            // Once we are an frame -122 or higher we are done getting match settings
-            // Tell the iterator to stop
-            return;
-        }
-        // handle settings calculation
-        if (payload.frame <= Frames.FIRST) {
-            const playerIndex = payload.playerIndex;
-            const playersByIndex = _.keyBy(this.settings.players, 'playerIndex');
-
-            switch (payload.internalCharacterId) {
-                case 0x7:
-                    playersByIndex[playerIndex].characterId = 0x13; // Sheik
-                    break;
-                case 0x13:
-                    playersByIndex[playerIndex].characterId = 0x12; // Zelda
-                    break;
-            }
-        }
-    }
-
-    public handleFrameUpdate(command: Command, payload: PreFrameUpdateType | PostFrameUpdateType): void {
-        payload = payload as PostFrameUpdateType;
-        if (!payload.frame && payload.frame !== 0) {
-            // If payload is messed up, stop iterating. This shouldn't ever happen
-            return;
-        }
-
-        const location = command === Command.PRE_FRAME_UPDATE ? "pre" : "post";
-        const frames = payload.isFollower ? this.followerFrames : this.playerFrames;
-        this.latestFrameIndex = payload.frame;
-        _.set(frames, [payload.frame, 'players', payload.playerIndex, location], payload);
-        _.set(frames, [payload.frame, 'frame'], payload.frame);
-
-        this.statsComputer.addFrame(frames[payload.frame]);
+  }
+
+  public handlePostFrameUpdate(payload: PostFrameUpdateType): void {
+    // Finish calculating settings
+    if (!this.settingsComplete && payload.frame <= Frames.FIRST) {
+      const playerIndex = payload.playerIndex;
+      const playersByIndex = _.keyBy(this.settings.players, 'playerIndex');
+
+      switch (payload.internalCharacterId) {
+      case 0x7:
+        playersByIndex[playerIndex].characterId = 0x13; // Sheik
+        break;
+      case 0x13:
+        playersByIndex[playerIndex].characterId = 0x12; // Zelda
+        break;
+      }
+      this.settingsComplete = true;
```

Comment:
This re-introduces a bug I had fixed when I changed how the short circuiting worked. Probably important to understand.

Settings are not complete (pre 1.6.0) until all of the post frame updates have come through for the first frame, not just the first post frame update. This is used to set the character properly in the case of a sheik transform. The way the code is written now, if sheik played in the higher port number, the settings would should them as Zelda.

`this.settingsComplete = payload.frame > Frames.FIRST` could be placed outside the if statement and it would work properly. This causes it to wait for the first frame update of frame -122, meaning that all -123 post frames have come through.

Reply:
Fixed in commit b801d6.